### PR TITLE
dvc metrics show: implement jsonpath-ng

### LINKED ISSUE #1919: dvc metrics show - implement jsonpath-ng instead of python-jsonpath-rw
Hello,

For passing JSONPath to ``dvc metrics show`` is it possible to use [jsonpath-ng](https://github.com/h2non/jsonpath-ng) under the hood instead of [python-jsonpath-rw](https://github.com/kennknowles/python-jsonpath-rw)?

``jsonpath-ng`` seems to more fully support JSONPath features, for example the ability to filter on certain values: ``$..book[?(@.price<10)]`` which would be useful for comparing metrics with greater specificity from dvc metrics files.

Thanks for considering!

### PR DESCRIPTION
**Pull request by brbarkley**
Fixes #1919

* [X] Have you followed the guidelines in our
      [Contributing document](https://dvc.org/doc/user-guide/contributing)?

* [ ] Does your PR affect documented changes or does it add new functionality
      that should be documented? If yes, have you created a PR for
      [dvc.org](https://github.com/iterative/dvc.org) documenting it or at
      least opened an issue for it? If so, please add a link to it.

-----

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -15,7 +15,7 @@ nanotime>=0.5.2
 pyasn1>=0.4.1
 schema>=0.6.7
 paramiko>=2.4.1
-jsonpath-rw==1.4.0
+jsonpath-ng>=1.4.3
```
Comment (efiop) [inline requirements.txt]: Need to update `setup.py` as well.

**Comment by efiop** [comment]
@brbarkley Looks great! Thank you! There is one test failing in https://github.com/iterative/dvc/blob/master/tests/func/test_metrics.py#L295 . You just need to add `metric_json_ext` to those lists and we should be all set :slightly_smiling_face:

File: tests/func/test_metrics.py [hunk 2]
```diff
@@ -101,6 +131,26 @@ def test_show(self):
         self.assertSequenceEqual(ret["bar"]["metric_hcsv"], ["bar"])
         self.assertSequenceEqual(ret["baz"]["metric_hcsv"], ["baz"])
 
+        ret = self.dvc.metrics.show(
+            "metric_json_ext",
+            typ="json",
+            xpath="$.metrics[?(@.deviation_mse<0.30) & (@.value_mse>0.4)]",
```
Comment (efiop) [inline tests/func/test_metrics.py]: Wow, this is truly something :slightly_smiling_face: We should definitely document this. Could you please adjust `--xpath` description for json in https://dvc.org/doc/commands-reference/metrics-modify ?

**Comment by efiop** [comment]
Thank you! :tada:

**Comment by brbarkley** [comment]
> Looks great! The only thing left to do is to adjust the docs.
See PR for docs [#275](https://github.com/iterative/dvc.org/pull/275). Obviously, ignore the PR that I accidentally submitted to my own fork 🙂

**Pull request merged**